### PR TITLE
Audio: Implmenting AudioOutReleasePort

### PR DESCRIPTION
**Pull request by tictakk**
Implemented audio out release and modified audio out open. Tested on Haiyore! If there are other games that this can be tested on let me know.

### CODE REVIEW

**Comment by Zangetsu38** [comment]
tips for you, my repo for log
https://github.com/Zangetsu38/Log-Vita3K/search?q=sceAudioOutReleasePort

**Comment by tictakk** [comment]
Awesome, I'll test some of these and see if there are any issues.

**Comment by Zangetsu38** [comment]
bad works for me, bad apple vita have die sound
image

**Comment by Zangetsu38** [comment]
this function is very used also maybe important 
sceAppMgrAcquireBgmPort

**Comment by Zangetsu38** [comment]
squash all, all time edit last commit and force push :)

**Comment by Zangetsu38** [comment]
run format, and edit commit name, remove other message inside ^^

File: vita3k/modules/SceAudio/SceAudio.cpp [hunk 1]
```diff
@@ -91,9 +91,20 @@ EXPORT(int, sceAudioOutOpenPort, SceAudioOutPortType type, int len, int freq, Sc
     port->shared.stream = stream;
 
     const std::lock_guard<std::mutex> lock(host.audio.shared.mutex);
-    const int port_id = host.audio.shared.next_port_id++;
+
+    //AudioOutPortPtrs ports = host.audio.shared.out_ports;
```
Comment (Zangetsu38) [inline vita3k/modules/SceAudio/SceAudio.cpp:95]: remove this if is not needed

**Comment by Zangetsu38** [comment]
good nown, squash, and set commit name one line :)

**Comment by Zangetsu38** [comment]
don't create new commit, change code, and edit last commit + force push

File: vita3k/modules/SceAudio/SceAudio.cpp [hunk 2]
```diff
@@ -91,9 +91,19 @@ EXPORT(int, sceAudioOutOpenPort, SceAudioOutPortType type, int len, int freq, Sc
...
 
+    //find next lowest available port
+    int highest_port_id = host.audio.shared.out_ports.rbegin()->first;
+    for (int i = 1; i < highest_port_id + 1; i++) {
+        if (!host.audio.shared.out_ports.find(i)->second) {
+            host.audio.shared.next_port_id = i;
+            return i;
+        }
+    }
+    host.audio.shared.next_port_id = port_id + 1;
```
Comment (Zangetsu38) [inline vita3k/modules/SceAudio/SceAudio.cpp:106]:
```suggestion
    host.audio.shared.next_port_id = port_id++;
```

Comment (tictakk) [reply vita3k/modules/SceAudio/SceAudio.cpp:106]: This throws an error for me when I tried it

Comment (Zangetsu38) [reply vita3k/modules/SceAudio/SceAudio.cpp:106]: oh, maybe ++port_id

Comment (Zangetsu38) [reply vita3k/modules/SceAudio/SceAudio.cpp:106]: ahnoà sorry, is const lol

Comment (tictakk) [reply vita3k/modules/SceAudio/SceAudio.cpp:106]: "expression must be modifiable lvalue"

Comment (tictakk) [reply vita3k/modules/SceAudio/SceAudio.cpp:106]: pour les deux

Comment (Zangetsu38) [reply vita3k/modules/SceAudio/SceAudio.cpp:106]: tes, i know no worry ^^

**Comment by Zangetsu38** [comment]
Is fine, so waiting review by other dev :)

**Comment by tictakk** [comment]
@pent0 could I get eyes on this so I can work on my next PR? ty

**Comment by pent0** [comment]
Consider my notes in the server :)

**Comment by pent0** [comment]
where are u ggggggggg

**Comment by tictakk** [comment]
@pent0 I'm back now, where are you???

**Comment by Zangetsu38** [comment]
@tictakk need rebase and fix conflict

**Comment by Zangetsu38** [comment]
outdated, and wrong code thx for your works

**Pull request closed without merging**